nepalingo-web/src/components/userAuth/AuthContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  ReactNode,
+  useEffect,
+} from "react";
+import supabase from "./supabaseClient";
+import { AuthError, AuthResponse, AuthTokenResponsePassword, SignInWithPasswordCredentials, SignUpWithPasswordCredentials, User } from "@supabase/supabase-js";
+
+interface AuthContextProps {
+  user: User | null;
+  signUp: (data: SignUpWithPasswordCredentials) => Promise<AuthResponse>;
+  signIn: (data: SignInWithPasswordCredentials) => Promise<AuthTokenResponsePassword>;
+  signOut: () => Promise<{ error: AuthError | null }> 
+}
+
+const AuthContext = createContext<AuthContextProps | undefined>(undefined);
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [, setIsAuthenticated] = useState(false);
```

Comment:
If you don't need isAuthenticated anymore - you can get rid of this. Since you can figure out whether the user is authenticated from the user data